### ISSUE TITLE
Guard against simplification of block which only includes variable definitions.

### ISSUE THREAD

**Issue by yossigil** [SpartanRefactoring/Spartanizer#154]
```java
	public void b(final boolean b1, final boolean b2) {
        if (b1)
			if (b2){ int i = 0;} else {int i = 0;}
    }
```
not to:

```java
	public void b(final boolean b1, final boolean b2) {
        if (b1)
			if (b2)int i = 0; else int i = 0;
    }
```
but rather to
```java
	public void b(final boolean b1, final boolean b2) {
    }
```

**Comment by OriRoth** [comment]
See issue #48, I have encountered a similar problem with IfBarFooElseBazFoo (common suffix):
```
if (b) {
  int i;
  i = 1;
  System.out.println(i);
} else {
  int i;
  i = 2;
  System.out.println(i);
}
```
I think there are some utility functions I made that could help.

**Comment by dormaayan** [comment]
@yossigil :
I wrote a code that removes the variable declarations in those cases and leaves an empty block.
There is a wring to remove conditions with empty block, but there isn't one to loops.
Do you want me to write wrings for removing loops in case there is an empty block or just leave it like that ? (Im aware to the fact that you won't necessarily want to remove the loop, for example when you do an action in the definition of the loop, so I just wandering what to do).

**Comment by dormaayan** [comment]
@yossigil :
also, after simplifying something like :
`trimming("{int a=0;}").to("int a=0;")`
It expects to :` "package p;public class C{public void m(){"`, Im not sure what is that mean (It doest look eve like a legal code), but, when I run the plugin it works just fine.
Do you want me to change all (the many) tests with only variable declarations to expect that ?

**Comment by OriRoth** [comment]
@dormaayan Its a problem created by running spartanization on the project. You can try reverting the specific file. Anyway, those tests in the current condition are indeed invalid.

**Comment by dormaayan** [comment]
@yossigil:
@OriRoth :
So, what do you suggest that i will do with that issue ?

**Comment by yossigil** [comment]
We need to fix the tests... OOPS. Mark this as won't fix, close, and disable the wring for now.

**Comment by yossigil** [comment]
Moved to backlog

**Comment by yossigil** [comment]
Are you @NivShalmon and @dormaayan  still working on this?

**Comment by yossigil** [comment]
?

**Comment by yossigil** [comment]
Related to #251 ? Pending on it?

**Comment by dormaayan** [comment]
@yossigil : issue #251 solved it
I close that issue

**Comment by yossigil** [comment]
:+1:
